peer/brontide.go

```diff
@@ -3133,19 +3133,18 @@ func (p *Brontide) tryLinkShutdown(cid lnwire.ChannelID) error {
 		return ErrChannelNotFound
 	}
 
-	// Else, the link exists, so attempt to trigger shutdown. If this
-	// fails, we'll send an error message to the remote peer.
-	if err := chanLink.ShutdownIfChannelClean(); err != nil {
-		return err
-	}
-
-	// Next, we remove the link from the switch to shut down all of the
-	// link's goroutines and remove it from the switch's internal maps. We
-	// don't call WipeChannel as the channel must still be in the
-	// activeChannels map to process coop close messages.
-	p.cfg.Switch.RemoveLink(cid)
-
-	return nil
+	return chanLink.Flush(func() {
+		// Else, the link exists, so attempt to trigger shutdown. If
+		// this fails, we'll send an error message to the remote peer.
+		chanLink.ShutdownHtlcManager()
+
+		// Next, we remove the link from the switch to shut down all of
+		// the link's goroutines and remove it from the switch's
+		// internal maps. We don't call WipeChannel as the channel must
+		// still be in the activeChannels map to process coop close
+		// messages.
+		p.cfg.Switch.RemoveLink(cid)
```

Comment:
IIUC, we need to insert the continuation signal here. In other words, in the case where we're trying to shutdown to do co-op close, the control flow (when we receive the request from our remote peer) needs to block _after_ sending `shutdown` (it shouldn't send `closing_signed` until the shutdown is finished). For the RPC flow, IMO we still want to just hard exit here as default behavior. We can add a new sync flow with a new API or arg to the existing API.